### ISSUE TITLE
Incorrect directive name processing when "vue-eslint-parser" used

### ISSUE THREAD

**Issue by evgeniymweb100** [vuejs/eslint-plugin-vue#948]
* **ESLint version: 6.2.1**	
* **eslint-plugin-vue version: 5.2.3**	
* **Node version: v12.2.0**	
* **vue-eslint-parser: 6.0.4**

```json
{
    "parser": "vue-eslint-parser",
    "parserOptions" : {
        "parser": "babel-eslint",
        "ecmaVersion" : 2018,
        "sourceType" : "module",
        "ecmaFeatures" : {
            "jsx" : true
        }
    },
    "extends" : [
        "plugin:vue/recommended",
        "plugin:import/errors",
        "plugin:import/warnings",
        "plugin:@typescript-eslint/recommended"
    ],
    "env": {
        "browser" : true,
        "node" : true
    },
    "rules" : {
        "array-bracket-newline" : [
            "warn", "consistent"
        ],
        "array-bracket-spacing" : ["warn", "never"],
        "array-callback-return" : [
            "error",
            {
                "allowImplicit" : true
            }
        ],
        "array-element-newline" : ["warn", "consistent"],
        "arrow-parens" : [
            "warn",
            "always",
            {
                "requireForBlockBody": true
            }
        ],
        "arrow-spacing" : [
            "warn",
            {
                "before" : true,
                "after" : true
            }
        ],
        "block-scoped-var" : "warn",
        "block-spacing" : ["warn", "always"],
        "brace-style" : ["error", "1tbs"],
        "camelcase" : [
            "warn",
            {
                "ignoreDestructuring" : true
            }
        ],
        "comma-dangle" : [
            "error",
            {
                "arrays" : "always-multiline",
                "objects" : "always-multiline",
                "imports" : "always-multiline",
                "exports" : "always-multiline",
                "functions" : "ignore"
            }
        ],
        "comma-spacing" : [
            "error",
            {
                "before" : false,
                "after" : true
            }
        ],
        "comma-style" : ["warn", "last"],
        "computed-property-spacing" : ["warn", "never"],
        "consistent-return" : "warn",
        "constructor-super" : "error",
        "curly" : ["error", "all"],
        "dot-notation" : [
            "warn",
            {
                "allowKeywords" : true,
                "allowPattern" :"^class$"
            }
        ],
        "eol-last" : ["error", "always"],
        "eqeqeq" : ["warn", "always"],
        "func-call-spacing" : ["error", "never"],
        "func-style" : ["warn", "expression"],
        "function-paren-newline" : ["error", "multiline"],
        "generator-star-spacing" : [
            "error",
            {
                "before" : false,
                "after" : true,
                "anonymous" : "neither",
                "method" : {
                    "before" : true,
                    "after" : true
                }
            }
        ],
        "guard-for-in" : "warn",
        "implicit-arrow-linebreak" : ["error", "beside"],
        "import/first" : "warn",
        "import/no-mutable-exports" : "warn",
        "import/no-unresolved" : "off",
        "import/no-webpack-loader-syntax" : "warn",
        "import/prefer-default-export" : "off",
        "indent" : [
            "error",
            4,
            {
                "ignoreComments": true,
                "SwitchCase": 1
            }
        ],
        "key-spacing" : [
            "error",
            {
                "beforeColon": false,
                "afterColon": true,
                "mode": "strict"
            }
        ],
        "keyword-spacing": "error",
        "linebreak-style" : ["error", "unix"],
        "new-parens" : "error",
        "no-alert" : "error",
        "no-array-constructor" : "error",
        "no-await-in-loop" : "warn",
        "no-caller" : "error",
        "no-case-declarations" : "warn",
        "no-class-assign" : "warn",
        "no-cond-assign" : ["error", "except-parens"],
        "no-confusing-arrow" : "error",
        "no-console" : [
            "error",
            {
                "allow" : ["warn", "error"]
            }
        ],
        "no-const-assign" : "error",
        "no-constant-condition" : "error",
        "no-debugger" : "error",
        "no-delete-var" : "warn",
        "no-div-regex" : "warn",
        "no-dupe-args" : "error",
        "no-dupe-class-members" : "error",
        "no-dupe-keys" : "error",
        "no-duplicate-case" : "error",
        "no-duplicate-imports" : "error",
        "no-else-return" : "warn",
        "no-empty" : [
            "error",
            {
                "allowEmptyCatch" : true
            }
        ],
        "no-empty-character-class" : "error",
        "no-empty-function" : "off",
        "no-empty-pattern" : "warn",
        "no-eq-null" : "error",
        "no-eval" : "error",
        "no-extra-bind" : "warn",
        "no-extra-boolean-cast" : "warn",
        "no-extra-parens" : "warn",
        "no-extra-semi" : "error",
        "no-floating-decimal" : "warn",
        "no-func-assign" : "error",
        "no-global-assign" : "warn",
        "no-implicit-coercion" : "warn",
        "no-implicit-globals" : "error",
        "no-implied-eval" : "error",
        "no-inner-declarations" : "warn",
        "no-invalid-regexp" : "error",
        "no-irregular-whitespace" : "warn",
        "no-iterator" : "error",
        "no-label-var" : "error",
        "no-labels" : "warn",
        "no-loop-func" : "warn",
        "no-mixed-spaces-and-tabs" : "error",
        "no-multi-assign" : "warn",
        "no-multi-spaces" : "error",
        "no-multi-str" : "warn",
        "no-multiple-empty-lines" : "error",
        "no-nested-ternary" : "warn",
        "no-new-func" : "error",
        "no-new-object" : "error",
        "no-new-symbol" : "error",
        "no-new-wrappers" : "warn",
        "no-obj-calls" : "error",
        "no-octal" : "warn",
        "no-octal-escape" : "warn",
        "no-proto" : "error",
        "no-prototype-builtins" : "error",
        "no-redeclare" : [
            "error",
            {
                "builtinGlobals" : true
            }
        ],
        "no-regex-spaces" : "error",
        "no-restricted-properties" : "warn",
        "no-restricted-syntax" : "error",
        "no-return-assign" : "warn",
        "no-return-await" : "warn",
        "no-self-assign" : "warn",
        "no-self-compare" : "warn",
        "no-sequences" : "warn",
        "no-shadow" : "warn",
        "no-shadow-restricted-names" : "warn",
        "no-sparse-arrays" : "warn",
        "no-tabs" : "error",
        "no-template-curly-in-string" : "warn",
        "no-this-before-super" : "warn",
        "no-throw-literal" : "warn",
        "no-trailing-spaces" : "error",
        "no-undef" : "error",
        "no-unexpected-multiline" : "error",
        "no-unmodified-loop-condition" : "warn",
        "no-unneeded-ternary" : "warn",
        "no-unreachable" : "warn",
        "no-unsafe-finally" : "error",
        "no-unsafe-negation" : "error",
        "no-unused-expressions" : "warn",
        "no-unused-labels" : "warn",
        "no-unused-vars" : "warn",
        "no-use-before-define" : "warn",
        "no-useless-call" : "warn",
        "no-useless-computed-key" : "error",
        "no-useless-concat" : "warn",
        "no-useless-constructor" : "warn",
        "no-useless-escape" : "warn",
        "no-useless-rename" : "warn",
        "no-useless-return" : "warn",
        "no-var" : "error",
        "no-void" : "warn",
        "no-whitespace-before-property" : "error",
        "no-with" : "error",
        "object-curly-newline" : [
            "error",
            {
                "ObjectExpression" : {
                    "multiline" : true,
                    "minProperties" : 2,
                    "consistent" : true
                }
            }
        ],
        "object-curly-spacing" : ["error", "always"],
        "object-property-newline" : "error",
        "object-shorthand" : ["warn", "always"],
        "one-var" : [
            "error",
            {
                "const" : "never",
                "let" : "never"
            }
        ],
        "operator-assignment" : ["warn", "always"],
        "operator-linebreak" : [
            "error",
            "after",
            {
                "overrides" : {
                    "=": "never"
                }
            }
        ],
        "padded-blocks" : ["error", "never"],
        "prefer-arrow-callback" : [
            "warn",
            {
                "allowNamedFunctions" : true
            }
        ],
        "prefer-numeric-literals" : "error",
        "prefer-object-spread" : "warn",
        "prefer-rest-params" : "error",
        "prefer-spread" : "warn",
        "prefer-template" : "warn",
        "quote-props" : ["warn", "as-needed"],
        "quotes" : ["warn", "single"],
        "radix" : ["error", "always"],
        "rest-spread-spacing" : ["error", "never"],
        "require-atomic-updates" : "warn",
        "require-await" : "warn",
        "require-yield" : "warn",
        "semi" : ["error", "always"],
        "semi-spacing" : [
            "error",
            {
                "before": false,
                "after": true
            }
        ],
        "sort-imports" : [
            "error",
            {
                "ignoreCase" : true,
                "ignoreMemberSort": true,
                "memberSyntaxSortOrder": ["none", "all", "single", "multiple"]
            }
        ],
        "space-before-blocks" : [
            "warn",
            {
                "functions" : "always",
                "keywords" : "always",
                "classes" : "always"
            }
        ],
        "space-before-function-paren" : [
            "error",
            {
                "anonymous" : "always",
                "named" : "never",
                "asyncArrow" : "always"
            }
        ],
        "space-in-parens" : ["error", "never"],
        "space-infix-ops" : "error",
        "space-unary-ops" : "error",
        "strict" : ["error", "global"],
        "symbol-description" : "warn",
        "switch-colon-spacing" : "error",
        "template-curly-spacing" : ["error", "never"],
        "unicode-bom" : ["error", "never"],
        "use-isnan" : "error",
        "valid-typeof" : [
            "error",
            {
                "requireStringLiterals" : true
            }
        ],
        "vue/html-indent" : [
            "error",
            4
        ],
        "vue/attribute-hyphenation" : 0,
        "vue/max-attributes-per-line" : [
            "error",
            {
                "singleline" : 3,
                "multiline" : {
                    "max" : 1,
                    "allowFirstLine" : false
                }
            }
        ],
        "vue/multiline-html-element-content-newline": [
            "error"
        ],
        "vue/singleline-html-element-content-newline": 0,
        "wrap-iife" : ["error", "inside"],
        "yoda" : ["error", "never"]
    },
}

```

**What did you do?**	

```vue
<template>
    <Comp
        aaa="aaa"
        :zzz="'zzz'"
        v-html="'qqq'"
    />
</template>

<script>
export default {
};
</script>

```

attributes-order rule checks an attribute type passing it name to function getAttributeType within function getPosition. The problem is that attribute.key.name is a string only for attributes, if attribute is a directive it is an object and needs to be accesses as attribute.key.name.name as well as attribute.key.argument should be attribute.key.argument.name for directives. As a result all v-bind "attributes" are detected as 'OTHER_DIRECTIVES' instead of 'OTHER_ATTR' and attribute order is broken.

This does not happen when "parser" option is not specified in top of config.

There are similar code snippets in some other rules.

**Comment by ota-meshi** [comment]
Thank you for this issue.

Since `eslint-plugin-vue@5.2.3` is not compatible with `vue-eslint-parser@6.0.4`, each rule may report incorrectly.
Please use the latest version of `eslint-plugin-vue@6.0.0` that was released the other day.
